examples/model_compress/pruning/taylorfo_pruning_with_ddp_torch.py

```diff
@@ -0,0 +1,257 @@
+from nni.compression.pytorch.utils import TorchEvaluator
+
+
+#############  Create dataloaders, optimizer, training and evaluation function ############
+
+class Mnist(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = torch.nn.Conv2d(1, 20, 5, 1)
+        self.conv2 = torch.nn.Conv2d(20, 50, 5, 1)
+        self.fc1 = torch.nn.Linear(4 * 4 * 50, 500)
+        self.fc2 = torch.nn.Linear(500, 10)
+        self.relu1 = torch.nn.ReLU6()
+        self.relu2 = torch.nn.ReLU6()
+        self.relu3 = torch.nn.ReLU6()
+        self.max_pool1 = torch.nn.MaxPool2d(2, 2)
+        self.max_pool2 = torch.nn.MaxPool2d(2, 2)
+
+    def forward(self, x):
+        x = self.relu1(self.conv1(x))
+        x = self.max_pool1(x)
+        x = self.relu2(self.conv2(x))
+        x = self.max_pool2(x)
+        x = x.view(x.shape[0], -1)
+        x = self.relu3(self.fc1(x))
+        x = self.fc2(x)
+        return x
+
+
+def create_dataloaders():
+    trans = transforms.Compose([transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))])
+    # training dataloader
+    training_dataset = datasets.MNIST('data', train=True, download=True, transform=trans)
+    training_sampler = torch.utils.data.distributed.DistributedSampler(training_dataset)
+    training_dataloader = torch.utils.data.DataLoader(training_dataset, \
+        batch_size=64, sampler=training_sampler)
+    # validation dataloader
+    validation_dataset = datasets.MNIST('data', train=False, transform=trans)
+    validation_sampler = torch.utils.data.distributed.DistributedSampler(validation_dataset)
+    validation_dataloader = torch.utils.data.DataLoader(validation_dataset, \
+        batch_size=1000, sampler=validation_sampler)
+
+    return training_dataloader, validation_dataloader
+
+def training(
+    training_dataloader: DataLoader,
+    model: nn.Module,
+    optimizer: torch.optim.Optimizer,
+    criterion: Callable[[torch.Tensor, torch.Tensor], torch.Tensor],
+    lr_scheduler: torch.optim.lr_scheduler._LRScheduler = None,
+    max_steps: int = None, max_epochs: int = None,
+    local_rank: int = -1,
+    save_best_model: bool = False, save_path: str = None,
+    log_path: str = None,
+    evaluation_func=None,
+):
+    model.train()
+    current_step = 0
+    best_acc = 0.
+
+    for current_epoch in range(max_epochs if max_epochs else 2):
+        for (data, target) in training_dataloader:
+            data, target = data.cuda(), target.cuda()
+            optimizer.zero_grad()
+            output = model(data)
+            loss = criterion(output, target)
+
+            loss.backward()
+            optimizer.step()
+
+            if lr_scheduler:
+                lr_scheduler.step()
+
+            current_step += 1
+
+        # evaluation for every 1000 steps
+        if current_step % 1000 == 0 or current_step % len(training_dataloader) == 0:
+            acc = evaluation_func(model)
+            with (log_path).open('a+') as f:
```

Comment:
why use `(log_path)`

Reply:
because it can be used to save some training information, such as the performance on the task-specific metrics